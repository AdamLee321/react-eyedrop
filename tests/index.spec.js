import React from 'react'
import { shallow, mount } from 'enzyme'
import App from '../src'
import rgbToHex from '../src/rgbToHex'

/**
 * passThrough
 * buttonComponent
 */

describe('React-EyeDrop should', () => {
  it('render', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  it('return colors to given callback', () => {
    const onChangeSpy = jest.fn()
    const instance = shallow(<App onChange={onChangeSpy} />).instance()
    instance.setColors({ r: 0, g: 0, b: 0 })
    expect(onChangeSpy).toHaveBeenCalledTimes(1)
    expect(onChangeSpy).toHaveBeenCalledWith({
      rgb: `rgb(0, 0, 0)`,
      hex: '#000000'
    })
  })

  it('use custom button component when given', () => {
    const customButton = () => <div className='custom-btn' style={{backgroundColor: 'red'}}>OMEGA BUTTON</div>
    const wrapper = shallow(<App buttonComponent={customButton} />)
    expect(wrapper.children().html()).toEqual("<div class=\"custom-btn\" style=\"background-color:red\">OMEGA BUTTON</div>")
  })

  it('pass down colors to custom button component if given prop', () => {
    const customButton = ({ colorObject: { rgb } }) => (
      <div className='custom-btn' style={{backgroundColor: rgb ? rgb : 'red'}}>OMEGA BUTTON</div>
    )
    const wrapper = shallow(<App onChange={() => {}} buttonComponent={customButton} passThrough={'colorObject'} />)
    expect(wrapper.children().html()).toEqual("<div class=\"custom-btn\" style=\"background-color:red\">OMEGA BUTTON</div>")
    wrapper.instance().setColors({ r: 0, g: 0, b: 0 })
    expect(wrapper.children().html()).toEqual("<div class=\"custom-btn\" style=\"background-color:rgb(0, 0, 0)\">OMEGA BUTTON</div>")
  })

  describe('set given classes:', () => {
    it('wrapper classes', () => {
      const wrapper = shallow(<App wrapperClasses="super-wrapper-class" />)
      expect(wrapper.find('div.super-wrapper-class').length).toBe(1)
    })
    it('button classes', () => {
      const wrapper = shallow(<App buttonClasses="mega-button-class" />)
      expect(wrapper.find('button.mega-button-class').length).toBe(1)
    })
  })

  describe('replace default value when given prop:', () => {
    it('once', () => {
      let instance = shallow(<App />).instance()
      expect(instance.once).toBe(true)
      instance = shallow(<App once={false} />).instance()
      expect(instance.once).toBe(false)
    })
    it('cursorActive', () => {
      let instance = shallow(<App />).instance()
      expect(instance.cursorActive).toBe('copy')
      instance = shallow(<App cursorActive='cursor' />).instance()
      expect(instance.cursorActive).toBe('cursor')
    })
    it('cursorInactive', () => {
      let instance = shallow(<App />).instance()
      expect(instance.cursorInactive).toBe('auto')
      instance = shallow(<App cursorInactive={'copy'} />).instance()
      expect(instance.cursorInactive).toBe('copy')
    })
  })

  describe('call given callback when the time is right:', () => {
    it('onInit', () => {
      const onInitSpy = jest.fn()
      const wrapper = shallow(<App onInit={onInitSpy} />)
      expect(onInitSpy).toHaveBeenCalledTimes(1)
    })
    it('onPickStart', () => {
      const onPickStartSpy = jest.fn()
      const instance = shallow(<App onPickStart={onPickStartSpy} />).instance()
      expect(onPickStartSpy).toHaveBeenCalledTimes(0)
      instance.pickColor()
      expect(onPickStartSpy).toHaveBeenCalledTimes(1)
    })
    it('onPickEnd', () => {
      const onPickEndSpy = jest.fn()
      const instance = shallow(<App onChange={() => {}} onPickEnd={onPickEndSpy} passThrough='hi' />).instance()
      expect(onPickEndSpy).toHaveBeenCalledTimes(0)
      instance.setColors({ r: 0, g: 255, b: 0 })
      expect(onPickEndSpy).toHaveBeenCalledTimes(1)
    })
  })
})

describe('rgbToHex should', () => {
  it('convert RGB color to HEX color', () => {
    const whiteRGB = { r: 255, g: 255, b: 255 }
    const { r, g, b } = whiteRGB

    const whiteHEX = rgbToHex(r, g, b)

    expect(whiteHEX).toBe('#ffffff')
  })
})