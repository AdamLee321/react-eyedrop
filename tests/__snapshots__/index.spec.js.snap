// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React-EyeDrop should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EyeDropper />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#000000",
            "border": "none",
            "borderRadius": "20%",
            "color": "#ffffff",
            "padding": "10px 25px",
          }
        }
      >
        Eye-Drop
      </button>,
      "className": undefined,
      "style": Object {
        "position": "relative",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Eye-Drop",
        "className": undefined,
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "#000000",
          "border": "none",
          "borderRadius": "20%",
          "color": "#ffffff",
          "padding": "10px 25px",
        },
      },
      "ref": null,
      "rendered": "Eye-Drop",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#000000",
              "border": "none",
              "borderRadius": "20%",
              "color": "#ffffff",
              "padding": "10px 25px",
            }
          }
        >
          Eye-Drop
        </button>,
        "className": undefined,
        "style": Object {
          "position": "relative",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Eye-Drop",
          "className": undefined,
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#000000",
            "border": "none",
            "borderRadius": "20%",
            "color": "#ffffff",
            "padding": "10px 25px",
          },
        },
        "ref": null,
        "rendered": "Eye-Drop",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
